app/controllers/posts_controller.rb

```diff
@@ -1,17 +1,12 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = Post.where(author_id: @user.id)
+    @posts = @user.posts
   end
 
   def show
-    @user = User.find(params[:user_id])
-
-    begin
-      @post = Post.where(author_id: @user.id).find(params[:id])
-    rescue StandardError
-      @post = nil
-    end
+    @post = Post.find(params[:id])
+    # @post = Post.includes(:author, :comments, :likes).find(params[:id])
```

Comment:
- [ ] Kindly remove the commented code since we should make our code clean and DRY and KISS. ♻🚀🔥